cmd/dp-filter-api/main.go

```diff
@@ -101,79 +113,125 @@ func main() {
 		cfg.EnablePrivateEndpoints,
 		cfg.DownloadServiceURL,
 		cfg.DownloadServiceSecretKey,
+		cfg.ServiceAuthToken,
 		auditor,
+		&hc,
 	)
 
-	go func() {
-		select {
-		case err := <-producer.Errors():
-			log.ErrorC("kafka producer error received", err, nil)
-		case err := <-auditProducer.Errors():
-			log.ErrorC("kafka audit producer error received", err, nil)
-		case err := <-apiErrors:
-			log.ErrorC("api error received", err, nil)
-		}
-	}()
-
 	// block until a fatal error occurs
 	select {
 	case <-signals:
-		log.Info("os signal received", nil)
+		log.Event(ctx, "os signal received")
```

Comment:
I think we can add the apierrors case back in here, if there is an issue with the server force a graceful shutdown as the application will be unable to process anything

Reply:
I agree - though in other services we have been just logging when a server error occurs, and not shutting down the service?

Reply:
Yes some because those applications have multiple responsibilities for instance import pipeline apps are predominantly event driven via kafka not by http requests, they are only have a http server for healthcheck endpoint.

If those apps http server fails then they maybe able to continue processing data and the platform can decide what to do (e.g. kill the service if it doesn't like the 500 response it will receive if the server is not working)

Reply:
In this case the API is reliant on having a server and therefore any issues should force the atempt to gracefully shutdown

Reply:
Makes sense - thanks for explaining 👍